test/extended/registry/registry.go

```diff
@@ -0,0 +1,158 @@
+package registry
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	dockerClient "github.com/fsouza/go-dockerclient"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	regclient "github.com/openshift/origin/pkg/dockerregistry"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imagesutil "github.com/openshift/origin/test/extended/images"
+	exutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+const (
+	repoName  = "app"
+	tagName   = "latest"
+	imageSize = 1024
+)
+
+var _ = g.Describe("[registry][migration] manifest migration from etcd to registry storage", func() {
```

Comment:
Add this test to the default test set in conformance and core (by adding a new reflex entry) so it always runs.